lib/bullhorn/users.ex

```diff
@@ -21,14 +36,33 @@ defmodule Bullhorn.Users do
     |> send_user_email(user)
   end
 
-  def created(%UserCreated{user: user}) do
-    user
-    |> UserEmails.welcome()
-    |> send_user_email(user)
+  def two_factor_requested(%TwoFactorRequest{token: token, user: user}) do
+    case user_notification_method(user, "two_factor") do
+      {:NOTIFICATION_METHOD_SMS, user} ->
+        Twilio.deliver_sms_two_factor_token(user, token)
+
+      {:NOTIFICATION_METHOD_VOICE, user} ->
+        Twilio.deliver_voice_two_factor_token(user, token)
+
+      _ ->
+        :ignored
+    end
   end
 
+  defp account_service_url, do: Application.get_env(:recognizer, :account_service_url)
+
   defp send_user_email(email, user) do
     Logger.metadata(recipient_id: user.id)
     Mailer.send(email)
   end
+
+  defp user_notification_method(user, event_type) do
+    request =
+      NotificationMethodRequest.new(event_type: event_type, request_id: Bottle.RequestId.write(:queue), user: user)
```

Comment:
@btkostner do we want to use `Bottle.RequestId.write(:queue)` for gRPC calls? If so, do we want to rename it to something other than `:queue`?

Reply:
There is an rpc type you can use instead

Reply:
TIL. Updated :tada: